src/core/Elsa.Abstractions/Models/WorkflowInstance.cs

```diff
@@ -33,13 +34,24 @@ public WorkflowInstance()
         public WorkflowInputReference? Input { get; set; }
         public WorkflowOutputReference? Output { get; set; }
         public IDictionary<string, IDictionary<string, object?>> ActivityData { get; set; } = new Dictionary<string, IDictionary<string, object?>>();
+        public IDictionary<string, object?> MetaData { get; set; } = new Dictionary<string, object?>();
 
         public HashSet<BlockingActivity> BlockingActivities
         {
             get => _blockingActivities;
             set => _blockingActivities = new HashSet<BlockingActivity>(value, BlockingActivityEqualityComparer.Instance);
         }
 
+        public object? GetMetaData(string key) { 
+            return MetaData.FirstOrDefault(x => x.Key == key).Value;

```

Comment:
Although we won't ever notice this is less efficient, the correct way to access values from a dictionary by key is by actually leveraging the dictionary instead of a linear lookup.

Like this:

```csharp
public object? GetMetaData(string key) => MetaData.TryGetValue(key, out var value) ? value : default;
```